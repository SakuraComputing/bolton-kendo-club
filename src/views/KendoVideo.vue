<template>
    <div class="kendo-video">
        <h1>Kendo Video</h1>
        <youtube 
            v-for="video in videos"
            :video-id="video.videoId"
            :key="video.etag"
        />
    </div>    
</template>

<script>
    import axios from 'axios';

    export default {
    data() {
        return {
            videos: []
        }
    },
    created: function() {
        this.fetchVideos();
    },
    methods: {
        fetchVideos() {
            axios.get('/api/club').then((response) => {
                this.videos = response.data[0].videos
            });
        }
    }
}
</script>

<style>

.kendo-video {
    margin: 8%;
}
</style>

