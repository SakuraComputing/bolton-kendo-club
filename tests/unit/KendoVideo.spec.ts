import { shallowMount } from '@vue/test-utils'
import KendoVideo from '@/views/KendoVideo.vue'

describe('KendoVideo.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(KendoVideo);
    expect(wrapper).toMatchSnapshot()
  })
})
