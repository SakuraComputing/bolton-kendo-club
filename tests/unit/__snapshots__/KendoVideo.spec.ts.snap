// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KendoVideo.vue renders correctly 1`] = `
<div class="kendo-video">
  <h1>Kendo Video</h1>
</div>
`;
